nexus/db-model/src/certificate.rs

```diff
@@ -8,8 +8,55 @@ use db_macros::Resource;
 use nexus_types::external_api::params;
 use nexus_types::external_api::views;
 use nexus_types::identity::Resource;
+use omicron_common::api::external::Error;
+use openssl::pkey::PKey;
+use openssl::x509::X509;
 use uuid::Uuid;
 
+#[derive(Debug, thiserror::Error)]
+pub enum CertificateError {
+    #[error("Failed to parse certificate: {0}")]
+    BadCertificate(openssl::error::ErrorStack),
+
+    #[error("Certificate exists, but is empty")]
+    CertificateEmpty,
+
+    #[error("Failed to parse private key")]
```

Comment:
Is there anything useful in the openssl stack here (that'd be useful to put into the error message the way we do for `BadCertificate`)?

Reply:
I had a request to do roughly the opposite of this in https://github.com/oxidecomputer/omicron/pull/2099#discussion_r1065046652 - I don't have any way to parse the openssl stack, and I haven't validated that it wouldn't leak private key info.

Reply:
Okay, we can deal with that if we run into this failure mode in the wild.